Objective-C/Internal/CBLDatabase+Internal.h

```diff
@@ -76,6 +78,9 @@ NS_ASSUME_NONNULL_BEGIN
 - (nullable NSString*) getCookies: (NSURL*)url error: (NSError**)error;
 - (BOOL) saveCookie: (NSString*)cookie url: (NSURL*)url;
 
+// Gets called from CBLCollection as well. Must be called inside a lock
+- (BOOL) prepareDocument: (CBLDocument*)document error: (NSError**)error;
```

Comment:
Can you double check if this method is used only from CBLCollection? If so, please move it to CBLCollection. My impression is that CBLDatabase class will not use this anymore as it will call into CBLCollection via the default collection.

Reply:
I thought I added a comment.
I have added a comment to remove this function, once we redirect the calls to Collection.